README.txt
This extension is still under heavy developement.
You have to patch vge_tagcloud for gh_multitag do work. Enter at your own risk...

http://forge.typo3.org/issues/show/3171

http://forge.typo3.org/issues/show/3173


ChangeLog
2009-10-19 Gregor Hermens <gregor@a-mazing.de>
	* adapted changes to the postProcessPages hook of vge_tagcloud
	* release 0.1.2

2009-10-18 Gregor Hermens <gregor@a-mazing.de>
	* some cleanup in documentation

2009-10-17 Gregor Hermens <gregor@a-mazing.de>
	* bugfix: no startpage transmitted when using styles on tags
	* release 0.1.1

2009-10-17 Gregor Hermens <gregor@a-mazing.de>
	* changed TS setup and urlen/decoding to work with vge_tagcloud 1.4.0
	* changed dependencies to vge_tagcloud >= 1.4.0
	* changed status to beta
	* some code cleanups
	* release 0.1.0

2009-04-27 Gregor Hermens <gregor@a-mazing.de>
	* added some documentation and screenshots
	* release 0.0.3

2009-04-27 Gregor Hermens <gregor@a-mazing.de>
	* already submitted tags are now urlencoded correctly
	* release 0.0.2

2009-04-27 Gregor Hermens <gregor@a-mazing.de>
	* alpha release 0.0.1

2009-04-24 Gregor Hermens <gregor@a-mazing.de>

	* initial code generated with kickstarter


vgetagcloud_hooks/class.tx_ghmultitag_vgetagcloud_hooks.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2009 Gregor Hermens <gregor@a-mazing.de>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 *
 *
 *   48: class tx_ghmultitag_vgetagcloud_hooks
 *   60:     function tx_ghmultitag_vgetagcloud_hooks()
 *   83:     function postProcessRawKeywords($keywords, &$callingObj)
 *  111:     function postProcessPages($pages, &$callingObj)
 *  132:     function processTagData(&$data)
 *
 * TOTAL FUNCTIONS: 4
 * (This index is automatically created/updated by the extension "extdeveval")
 *
 */


/**
 * This class uses hooks of tx_vgetagcloud_pi1 to enable search results based on more than one tag
 *
 * @author		Gregor Hermens <gregor@a-mazing.de>
 * @package		TYPO3
 * @subpackage	tx_ghmultitag
 */
class tx_ghmultitag_vgetagcloud_hooks {

	var $submittedKeywords = array();
	var $submittedPages = array();

	/**
	 * Constructor
	 *
	 * Fetches url parameters submitted by tx_vgetagcloud_pi1 for use in other methods
	 *
	 * @return	[type]		...
	 */
	function tx_ghmultitag_vgetagcloud_hooks() {
		$gpVars = t3lib_div::_GP('tx_vgetagcloud_pi2');
		$keywords = $gpVars['keyword'];
		if(!empty($keywords)) {
			$this->submittedKeywords = t3lib_div::trimExplode('_', $keywords,1);
		}

		$pages = $gpVars['pages'];
		if(!empty($pages) and preg_match('|^[0-9_]+$|', $pages)) {
			$this->submittedPages = t3lib_div::trimExplode('_', $pages,1);
		}
	}

	/**
	 * This method receives an array containing all the keywords found by the vge_tagcloud extension
	 * before any processing has been done on the keywords, except for case conversion
	 *
	 * All keywords already included into the search are removed from this list
	 *
	 * @param	array		$keywords: list of keywords
	 * @param	mixed		$callingObj: callback to the calling object
	 * @return	array		transformed list of keywords
	 */
	function postProcessRawKeywords($keywords, &$callingObj) {

		// Only process keywords if we are on the targetPage
		if($callingObj->conf['targetPage'] != $callingObj->cObj->data['uid']) {
			return $keywords;
		}

		if(empty($this->submittedKeywords)) {
			return $keywords;
		}

		$transformedKeywords = array();
		foreach ($keywords as $aKeyword) {
			if (!in_array($aKeyword, $this->submittedKeywords)) {
				$transformedKeywords[] = $aKeyword;
			}
		}
		return $transformedKeywords;
	}

	/**
	 * This method recieves an array of pages, from which the tags for the cloud are extracted.
	 * If a list of pages has already been submitted the original list is replaced by this one.
	 *
	 * @param	string		$pages: comma separated list of pages
	 * @param	mixed		$callingObj: callback to the calling object
	 * @return	string		transformed list of pages
	 */
	function postProcessPages($pages, &$callingObj) {

		// Only process pages if we are on the targetPage
		if($callingObj->conf['targetPage'] != $callingObj->cObj->data['uid']) {
			return $pages;
		}

		if(empty($this->submittedPages)) {
			return $pages;
		}
		return $this->submittedPages;
	}

	/**
	 * This method receives the data array of the calling cObj as a reference
	 * It can thus modify it as desired
	 *
	 * @param	array		$data: data array of the calling cObj
	 * @return	void
	 */
	function processTagData(&$data) {

		if(empty($this->submittedKeywords)) {
			return true;
		}

		$keywords = $this->submittedKeywords;
		$keywords[] = urldecode($data['tag_keyword']);
		$data['tag_keyword'] = implode('_', $keywords);
		return true;
	}
}


if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/gh_multitag/vgetagcloud_hooks/class.tx_ghmultitag_vgetagcloud_hooks.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/gh_multitag/vgetagcloud_hooks/class.tx_ghmultitag_vgetagcloud_hooks.php']);
}
?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "gh_multitag"
#
# Auto generated 19-10-2009 00:29
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Multi Tag Search for vge_tagcloud',
	'description' => 'Extends vge_tagcloud with a search using more than one tag.',
	'category' => 'plugin',
	'author' => 'Gregor Hermens',
	'author_email' => 'gregor@a-mazing.de',
	'shy' => '',
	'dependencies' => 'vge_tagcloud',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => '',
	'version' => '0.1.2',
	'constraints' => array(
		'depends' => array(
			'vge_tagcloud' => '1.4.0-0.0.0',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:12:{s:9:"ChangeLog";s:4:"57ec";s:10:"README.txt";s:4:"85bd";s:12:"ext_icon.gif";s:4:"a9c5";s:17:"ext_localconf.php";s:4:"c912";s:14:"ext_tables.php";s:4:"76e8";s:13:"locallang.xml";s:4:"16f7";s:59:"vgetagcloud_hooks/class.tx_ghmultitag_vgetagcloud_hooks.php";s:4:"068d";s:20:"static/constants.txt";s:4:"d41d";s:16:"static/setup.txt";s:4:"e53b";s:14:"doc/manual.sxw";s:4:"0f52";s:19:"doc/wizard_form.dat";s:4:"9f2f";s:20:"doc/wizard_form.html";s:4:"1c5d";}',
	'suggests' => array(
	),
);

?>